lib/ecto/enum.ex

```diff
@@ -71,4 +89,16 @@ defmodule Ecto.Enum do
 
   @impl true
   def embed_as(_, _), do: :self
+
+  def values(schema, field) do
+    try do
+      case schema.__schema__(:type, field) do
+        {:parameterized, Ecto.Enum, %{values: values}} -> values
+        {_, {:parameterized, Ecto.Enum, %{values: values}}} -> values
+        nil -> raise ArgumentError, "#{field} is not an Ecto.Enum field"
```

Comment:
This will be rescued and reraised. Seee suggestion below on a different way to write this!